lib/record_store/zone.rb

```diff
@@ -108,8 +112,52 @@ def write(**write_options)
       self.class.write(name, config: config, records: records, **write_options)
     end
 
+    ROOT_SERVERS = %w(
+      a.root-servers.net
+      b.root-servers.net
+      c.root-servers.net
+      d.root-servers.net
+      e.root-servers.net
+      f.root-servers.net
+      g.root-servers.net
+      h.root-servers.net
+      i.root-servers.net
+      j.root-servers.net
+      k.root-servers.net
+      l.root-servers.net
+      m.root-servers.net
+    )
+
+    def fetch_authority(nameserver = ROOT_SERVERS.sample)
+      Resolv::DNS.open(nameserver: nameserver) do |resolv|
+        resolv.fetch_resource(name, Resolv::DNS::Resource::IN::SOA) do |reply, name|
+          break if reply.answer.any?
```

Comment:
Does using a `return` not suffice here? How does this differ?

Reply:
rubocop complains about using return, recommending break instead